### DISCUSSION TITLE
Deploying and upgrading a UUPS proxy

### DISCUSSION THREAD

**Discussion by giladHaimov** [NomicFoundation/hardhat#5250] [General]
The documentation for Hardhat Deploy and update of a UUPS proxy is somewhat lacking.

Can anyone from the dev team confirm that using the standard deployProxy/upgradeProxy functions with an additional <kind : "uups"> opt is the correct way to handle a UUPS proxy.

That is:

```
//deploy a uups proxy:
    const _proxy = await upgrades.deployProxy(OrigContractFactory, [initialize -params-here], { 
              initializer: "initialize",
              kind : "uups",
            });



//upgrade a uups proxy:
    const _proxy = await upgrades.upgradeProxy(ORIG_PROXY_ADDRESS, NewContractFactory,  {kind : "uups"});

```

Thanks.

**Comment by kanej** [comment]
Are you using the Open Zeppelin upgrades plugin: https://docs.openzeppelin.com/upgrades-plugins/1.x/ ?

**Reply by giladHaimov** [reply]
@kanej indeed. I'm using OpenZeppelin upgrades plugin for Hardhat: https://docs.openzeppelin.com/upgrades-plugins/1.x/hardhat-upgrades

**Reply by kanej** [reply]
The Open Zeppelin team maintains that plugin and api but the docs indicate that `{kind: "uups"}` is a valid property of the `opts` object:

https://docs.openzeppelin.com/upgrades-plugins/1.x/api-hardhat-upgrades#deploy-proxy

**Reply by giladHaimov** [reply]
Thanks. see new thread [here](https://forum.openzeppelin.com/t/hardhat-upgrades-plugin-how-to-deploy-and-upgrade-a-uups-proxy/40581)
